guides/source/association_basics.md

```diff
@@ -793,15 +831,51 @@ class Book < ApplicationRecord
 end
 ```
 
-Active Record will no longer automatically recognize the bi-directional association:
+Because of the `:foreign_key` option Active Record will no longer automatically
+recognize the bi-directional association. This can cause your application to:
+
+* Execute needless queries for the same data (in this example causing N+1 queries)
 
 ```irb
-irb> a = Author.first
-irb> b = a.books.first
-irb> a.first_name == b.writer.first_name
+irb> author = Author.first
+irb> author.books.any? do |book|
+irb>   book.author.equal?(author) # This executes an author query for every book
+irb> end
+=> false
+```
+
+* Reference multiple copies of a model with inconsistent data
+
+```irb
+irb> author = Author.first
+irb> book = author.books.first
+irb> author.name == book.author.name
+=> true
+irb> author.name = 'Changed Name'
+irb> author.name == book.author.name
+=> false
+```
+
+* Fail to autosave associations
+
+```irb
+irb> author = Author.new
+irb> book = author.books.new
+irb> book.save!
+irb> book.persisted?
 => true
-irb> a.first_name = 'David'
-irb> a.first_name == b.writer.first_name
+irb> author.persisted?
+=> false
+```
+
+* Fail to validate presence or absence
+
+```irb
+irb> author = Author.new
+irb> book = author.books.new
+irb> book.author.present?
+=> false
+irb> book.valid?
 => false
```

Comment:
If you do update the snippet from line 806 to show validation error messages, this snippet should be updated to match.